src/main/java/org/opentripplanner/gtfs/mapping/StopAreaMapper.java

```diff
@@ -0,0 +1,73 @@
+package org.opentripplanner.gtfs.mapping;
+
+import static org.opentripplanner.gtfs.mapping.AgencyAndIdMapper.mapAgencyAndId;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import org.onebusaway.gtfs.model.Location;
+import org.onebusaway.gtfs.model.Stop;
+import org.opentripplanner.framework.collection.MapUtils;
+import org.opentripplanner.framework.i18n.NonLocalizedString;
+import org.opentripplanner.transit.model.site.GroupStop;
+import org.opentripplanner.transit.model.site.GroupStopBuilder;
+import org.opentripplanner.transit.service.StopModelBuilder;
+
+/**
+ * For a while GTFS Flex location groups were replaced by GTFS Fares v2 stop areas. After a few
+ * months, this decision was reverted and a new style of location groups we re-added to the Flex
+ * spec.
+ * <p>
+ * Arcadis tooling still produces stop areas and for a while we will support both. Please don't rely
```

Comment:
you can mark this sentence with the Javadoc tag `@deprecated` since it explains why the class is marked as `@Deprecated`